sklearn/metrics/pairwise.py

```diff
@@ -687,7 +613,245 @@ def cosine_distances(X, Y=None):
     return S
 
 
+def gower_distances(X, Y=None, categorical_features=None, scale=True, **metric_kwargs):
+    """Compute the distances between the observations in X and Y,
+    that may contain mixed types of data, using an implementation
+    of Gower formula.
+
+    Parameters
+    ----------
+    X : array-like, or pandas.DataFrame, shape (n_samples, n_features)
+
+    Y : array-like, or pandas.DataFrame, optional,
+        shape (n_samples, n_features)
+
+    categorical_features : array-like, optional, shape (n_features)
+        Indicates with True/False whether a column is a categorical attribute.
+        This is useful when categorical atributes are represented as integer
+        values. Categorical ordinal attributes are treated as numeric, and
+        must be marked as false.
+
+        Alternatively, the categorical_features array can be represented only
+        with the numerical indexes of the categorical attribtes.
+
+        If the categorical_features array is not provided, by default all
+        non-numeric columns are considered categorical.
+
+    scale : boolean, optional (default=True)
+        Indicates if the data will be scaled between 0 and 1. If false, it
+        is assumed the data is already scaled.
+
+    **metric_kwargs : dict, optional
```

Comment:
Having a generic keyword args is not a good idea. It's better to explicitly have MIN and MAX parameters although I argue against those particular params too.

Reply:
Makes sense, let's put it as an array to scale parameter.